pennylane/vqe/vqe.py

```diff
@@ -120,6 +120,32 @@ def __str__(self):
 
         return "\n+ ".join(terms)
 
+    @property
+    def is_diagonal(self):
+        r""" Checks if a Hamiltonian is diagonal in the computational basis.
+
+        Returns:
+            bool: ``True`` if the Hamiltonian is diagonal in the computational basis, ``False`` otherwise
+        """
+
+        diagonals = ["PauliZ", "Identity"]
+
+        non_diagonal_coeffs = []
+        non_diagonal_gates = []
+
+        for i, term in enumerate(self._ops):
+            gates = [term.name] if isinstance(term.name, str) else term.name
+            for j in gates:
+                if j not in diagonals:
+                    if gates not in non_diagonal_gates:
+                        non_diagonal_gates.append(gates)
+                        non_diagonal_coeffs.append(self._coeffs[i])
+                    else:
+                        non_diagonal_coeffs[non_diagonal_gates.index(gates)] += self._coeffs[i]
+                    break
+
+        return all([not bool(i) for i in non_diagonal_coeffs]) or len(non_diagonal_coeffs) == 0
```

Comment:
So what happens here if the non-diagonal coefficients are very small but non-zero?

Reply:
The function would return `False`. I'm assuming you're concerned about floating-point errors? If so, we can set some kind of tolerance 🙂

Reply:
Done!